neurom/core/neuron.py

```diff
@@ -156,3 +157,38 @@ def i_neurites(self, iterator_type, mapping=None, tree_filter=None):
             tree_filter: optional top level filter on properties of neurite tree objects.
         '''
         return tr.i_chain(self.neurites, iterator_type, mapping, tree_filter)
+
+
+def bounding_box(nrn):
+    '''Return 3D bounding box of a neuron
+
+    Returns:
+        tuple of ((min_x, min_y, min_z), (max_x, max_y, max_z))
+    '''
+
+    # Get the bounding coordinates of the neurites
+    nmin_xyz = (min(c for c in tr.i_chain(nrn.neurites, tr.ipreorder,
```

Comment:
I'm not sure the cost of making a numpy array is near the same as iterating so much.  Also, does the iteration make reference or return a new np array?  Finally, you can save one creation by doing:
np.minimum(min_value, pt, min_value)